olp-cpp-sdk-dataservice-read/tests/VersionedLayerClientOfflineTest.cpp

```diff
@@ -0,0 +1,407 @@
+ * License-Filename: LICENSE
+ */
+
+#include <gmock/gmock.h>
+#include <mocks/NetworkMock.h>
+#include <olp/authentication/Settings.h>
+#include <olp/core/client/OlpClientSettings.h>
+#include <olp/core/client/OlpClientSettingsFactory.h>
+#include <olp/core/http/NetworkRequest.h>
+#include <olp/core/http/NetworkResponse.h>
+#include <olp/core/logging/Log.h>
+#include <olp/core/porting/make_unique.h>
+#include <olp/dataservice/read/VersionedLayerClient.h>
+
+#include <chrono>
+#include <string>
+
+using namespace olp::dataservice::read;
+using namespace testing;
+
+namespace {
+
+constexpr char kHttpResponseLookupQuery[] =
+    R"jsonString([{"api":"query","version":"v1","baseURL":"https://query.data.api.platform.here.com/query/v1/catalogs/hereos-internal-test-v2","parameters":{}}])jsonString";
+
+constexpr char kHttpResponsePartition_269[] =
+    R"jsonString({ "partitions": [{"version":4,"partition":"269","layer":"testlayer","dataHandle":"4eed6ed1-0d32-43b9-ae79-043cb4256432"}]})jsonString";
+
+constexpr char kHttpResponseLookupBlob[] =
+    R"jsonString([{"api":"blob","version":"v1","baseURL":"https://blob-ireland.data.api.platform.here.com/blobstore/v1/catalogs/hereos-internal-test-v2","parameters":{}}])jsonString";
+
+constexpr char kHttpResponseBlobData_269[] =
+    R"jsonString(DT_2_0031)jsonString";
+
+constexpr auto kWaitTimeout = std::chrono::seconds(10);
+
+}  // namespace
+
+class VersionedLayerClientOfflineTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    network_mock_ = std::make_shared<NetworkMock>();
+
+    settings_ = std::make_shared<olp::client::OlpClientSettings>();
+    settings_->network_request_handler = network_mock_;
+  }
+
+  void TearDown() override {
+    auto network = std::move(settings_->network_request_handler);
+    network_mock_.reset();
+    settings_.reset();
+    // when test ends we must be sure that network pointer is not captured
+    // anywhere
+    ASSERT_EQ(network.use_count(), 1);
+  }
+
+  std::string getArgument(const std::string& name) {
+    if (name == "dataservice_read_test_catalog") {
+      return "hrn:here:data:::here-optimized-map-for-visualization-2";
+    } else if (name == "dataservice_read_test_layer") {
+      return "omv-base-v2";
+    } else if (name == "dataservice_read_test_partition") {
+      return "377804374";
+    } else if (name == "dataservice_read_test_layer_version") {
+      return "108";
+    }
+    ADD_FAILURE() << "unknown argument!";
+    return "";
+  }
+
+ protected:
+  std::shared_ptr<olp::client::OlpClientSettings> settings_;
+  std::shared_ptr<NetworkMock> network_mock_;
+};
+
+TEST_F(VersionedLayerClientOfflineTest, GetDataFromPartitionAsync) {
+  settings_->task_scheduler =
+      olp::client::OlpClientSettingsFactory::CreateDefaultTaskScheduler(1);
+
+  EXPECT_CALL(*network_mock_, Send(_, _, _, _, _))
+      .WillOnce(NetworkMock::ReturnHttpResponse(
+          olp::http::NetworkResponse().WithStatus(200),
```

Comment:
```suggestion
          olp::http::NetworkResponse().WithStatus(olp::http::HttpStatusCode::OK),
```
and in other places as well.